skimage/_shared/utils.py

```diff
@@ -547,6 +680,59 @@ def wrapped(*args, **kwargs):
         return wrapped
 
 
+def _deprecate_estimate(func, class_name=None):
+    """Deprecate ``estimate`` method."""
+    class_name = func.__qualname__.split('.')[0] if class_name is None else class_name
+    return deprecate_func(
+        deprecated_version="0.26",
+        removed_version="2.2",
```

Comment:
```suggestion
        removed_version="2.2",
```

Should this be 2.0?

Reply:
I was hoping to reduce the need-to-change-now footprint for 2.0.   What do you think?

Reply:
It makes probably more sense to schedule this for "2.0 (or later)". That's what I've been proposing in other PRs. It aligns user expectation with the big transition. I think message is "if you deal with deprecations slated for 2.0" you will only have a few breaking changes to worry about. 

We'll likely have to keep pre-v2.0 around on maintenance for quite some time.

Reply:
Do we agree we resolved this in the meeting a few days ago?

Reply:
I don't exactly remember. I think you advocated for drawing the deprecation out? It feels a bit weird to me to keep this working beyond 2.0. I'd prefer it saying "2.0 (or later)", but I can live with it. We will probably discuss this at least once a gain before the next release. :sweat_smile:

Reply:
Note, every deprecation should be noted in our `TODO.txt` so that we remember it. I'll add it.

Reply:
0.28 or 2.2, whichever comes earlier? :thinking: Quite hard to use post-2.0 numbers, since we don't know in how many versions 2.0 will be tagged.